recipes/xapian-core/all/conanfile.py

```diff
@@ -59,91 +60,103 @@ def requirements(self):
 
     def validate(self):
         if self.options.shared and self.settings.os == "Windows":
-            raise ConanInvalidConfiguration("shared builds are unavailable due to libtool's inability to create shared libraries")
+            raise ConanInvalidConfiguration(
+                "shared builds are unavailable due to libtool's inability to create shared libraries"
+            )
 
     def build_requirements(self):
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
+        if self._settings_build.os == "Windows":
+            self.win_bash = True
+            if not self.conf.get("tools.microsoft.bash:path", check_type=str):
+                self.tool_requires("msys2/cci.latest")
 
     def source(self):
-        get(self, **self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    @contextmanager
-    def _build_context(self):
-        if is_msvc(self):
-            with tools.vcvars(self.settings):
-                msvc_cl_sh =  f"{self.build_folder}/msvc_cl.sh".replace("\\", "/")
-                env = {
-                    "AR": "lib",
-                    "CC": msvc_cl_sh,
-                    "CXX": msvc_cl_sh,
-                    "LD": msvc_cl_sh,
-                    "NM": "dumpbin -symbols",
-                    "OBJDUMP": ":",
-                    "RANLIB": ":",
-                    "STRIP": ":",
-                }
-                with tools.environment_append(env):
-                    yield
-        else:
-            yield
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+        if not cross_building(self):
+            env = VirtualRunEnv(self)
+            env.generate(scope="build")
```

Comment:
I dont see this in out her recipes using `msys` why is it needed here? 🧐

Reply:
Sorry I don't understand. It's a very common pattern and it has nothing to do with msys: https://github.com/conan-io/conan-center-index/blob/811ff3245bcc1fa243672e4e469076d1feb5aa06/docs/package_templates/autotools_package/all/conanfile.py#L105-L109